reference/operators/query-jsonb-v2.md

```diff
@@ -103,6 +103,156 @@ SELECT jsonb_pretty(record) FROM logs WHERE record &@~ 'server OR mail';
+--  Bitmap Heap Scan on public.logs  (cost=0.00..21.03 rows=1 width=32) (actual time=1.577..1.578 rows=1 loops=1)
+--    Output: jsonb_pretty(record)
+--    Recheck Cond: (logs.record &@~ 'get'::text)
+--    Heap Blocks: exact=1
+--    ->  Bitmap Index Scan on pgroonga_logs_index  (cost=0.00..0.00 rows=14 width=0) (actual time=1.566..1.566 rows=1 loops=1)
+--          Index Cond: (logs.record &@~ 'get'::text)
+--  Planning Time: 0.680 ms
+--  Execution Time: 1.631 ms
+-- (8 rows)
+
+
+-- But when you search on specific key value in jsonb column, it DOES NOT USE Index 
+EXPLAIN ANALYZE verbose SELECT jsonb_pretty(record) FROM logs WHERE record->'message' &@~ 'get';
+--                                                QUERY PLAN                                                
+-- ---------------------------------------------------------------------------------------------------------
+--  Seq Scan on public.logs  (cost=0.00..1047.00 rows=1 width=32) (actual time=0.422..0.566 rows=1 loops=1)
+--    Output: jsonb_pretty(record)
+--    Filter: ((logs.record -> 'message'::text) &@~ 'get'::text)
+--    Rows Removed by Filter: 2
+--  Planning Time: 0.035 ms
+--  Execution Time: 0.576 ms
+-- (6 rows)
+```
+
+Now let's create an index of “message” key value in your record jsonb column: 
+```sql
+-- Create "message" key value index for your record jsonb column
+CREATE INDEX pgroonga_message_index ON logs USING pgroonga ((record->'message'));
+
+-- Before you execute EXPLAIN ANALYZE, you need to set sequential scan off
+-- to make sure it uses pgroonga index.
+-- Note: Do not set enable_seqscan = off on production environment
+SET enable_seqscan = off;
+
+-- Analyze the query: Now it uses pgroonga index
+EXPLAIN ANALYZE verbose SELECT jsonb_pretty(record) FROM logs WHERE record->'message' &@~ 'get';
+--                                                               QUERY PLAN                                                               
+-- ---------------------------------------------------------------------------------------------------------------------------------------
+--  Index Scan using pgroonga_message_index on public.logs  (cost=0.00..4.01 rows=1 width=32) (actual time=2.389..2.393 rows=1 loops=1)
+--    Output: jsonb_pretty(record)
+--    Index Cond: ((logs.record -> 'message'::text) &@~ 'get'::text)
+--  Planning Time: 0.201 ms
+--  Execution Time: 2.496 ms
+-- (5 rows)
+
+```
+
+
+## If you don’t know which jsonb key value should be indexed beforehand, then you should use `` &` `` operator instead
+Because of the nature of unstructured data, like json/jsonb data in this case, it is sometimes difficult to specify which key values to be indexed beforehand. (Say you need to store some user’s input which you don’t know what kind of data structure they use, and later user requests you to implement a search feature for that data.)
```

Comment:
```suggestion

Because of the nature of unstructured data, like json/jsonb data in this case, it is sometimes difficult to specify which key values to be indexed beforehand. (Say you need to store some user’s input which you don’t know what kind of data structure they use, and later user requests you to implement a search feature for that data.)
```